snapcraft/plugins/gulp.py

```diff
@@ -99,6 +99,14 @@ def build(self):
         env = os.environ.copy()
         env['PATH'] = '{}:{}'.format(
             os.path.join(self._npm_dir, 'bin'), env['PATH'])
+
+        # make npm use http proxy to access registry via https
+        # ref. https://github.com/npm/npm/issues/2050
+        if 'http_proxy' in os.environ:
```

Comment:
Only needs "`http_proxy`".

npm honors `http_proxy`, `https_proxy` and `no_proxy` environment variables. Normally if these variables are set, not need to use "`npm config`".  However if "`https_proxy`" is "https://proxy/", npm connect to registry server via HTTPS protocol and occur connection failed.

To workaround this, make npm use HTTP protocol by setting `https-proxy` to "http://proxy".

* `http_proxy`: use to access "http://registry.npmjs.org/"
* `https_proxy`: not use to access npm servers
* `no_proxy`: npm itself honors this environment variable